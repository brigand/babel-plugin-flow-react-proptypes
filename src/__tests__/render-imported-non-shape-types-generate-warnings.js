const content = `
import type {StringType, NumberType, EnumStringType} from './test-types.js.transpiled';

import React from 'react';
import renderer from 'react-test-renderer';

type FooProps = {
  stringValue: StringType,
  numberValue: NumberType,
  enumStringValue: EnumStringType,
};

class C extends React.Component {
    props: FooProps
    
    render() { return <div/> };
}
`;

const utils = require('./lib/render-component');

// TODO: this incorrectly generates no errors. The types are not marked as required!
it('imported-non-shape-generates-warnings: no props given', () => {
  const path = require('path');
  const renderCall = 'renderer.create(<C/>);';
  const fullSource = content + renderCall;
  const errorsSeen = utils.getConsoleErrorsForComponent(fullSource, [
    [path.join(__dirname, 'lib', 'test-types.js.pre-transpile'),
      path.join(__dirname, 'lib', 'test-types.js.transpiled')]
  ]);
  expect(errorsSeen).toMatchSnapshot();
});


it('imported-non-shape-generates-warnings: incorrect types given', () => {
  const path = require('path');
  const renderCall = 'renderer.create(<C stringValue={5} numberValue="string" enumStringValue="incorrect"/>);';
  const fullSource = content + renderCall;
  const errorsSeen = utils.getConsoleErrorsForComponent(fullSource, [
    [path.join(__dirname, 'lib', 'test-types.js.pre-transpile'),
      path.join(__dirname, 'lib', 'test-types.js.transpiled')]
  ]);
  expect(errorsSeen).toMatchSnapshot();
});

it('imported-non-shape-generates-warnings: correct types, no warning types given', () => {
  const path = require('path');
  const renderCall = 'renderer.create(<C stringValue="string" numberValue={5} enumStringValue="yes"/>);';
  const fullSource = content + renderCall;
  const errorsSeen = utils.getConsoleErrorsForComponent(fullSource, [
    [path.join(__dirname, 'lib', 'test-types.js.pre-transpile'),
      path.join(__dirname, 'lib', 'test-types.js.transpiled')]
  ]);
  expect(errorsSeen).toMatchSnapshot();
});
