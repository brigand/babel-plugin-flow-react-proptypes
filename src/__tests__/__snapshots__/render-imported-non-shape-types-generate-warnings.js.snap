// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`imported-non-shape-generates-warnings: correct types, no warning types given 1`] = `Array []`;

exports[`imported-non-shape-generates-warnings: incorrect types given 1`] = `
Array [
  Array [
    "Warning: Failed prop type: Invalid prop \`stringValue\` of type \`number\` supplied to \`C\`, expected \`string\`.
    in C",
  ],
  Array [
    "Warning: Failed prop type: Invalid prop \`numberValue\` of type \`string\` supplied to \`C\`, expected \`number\`.
    in C",
  ],
  Array [
    "Warning: Failed prop type: Invalid prop \`enumStringValue\` of value \`incorrect\` supplied to \`C\`, expected one of [\\"yes\\",\\"no\\"].
    in C",
  ],
]
`;

exports[`imported-non-shape-generates-warnings: no props given 1`] = `Array []`;
